pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -162,50 +198,59 @@ func (c *Cluster) updatePVCIfChanged(desiredPVC *v1.PersistentVolumeClaim, curre
 	}
 }
 
-func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, setIndex int, pvcTemplate v1.PersistentVolumeClaim) *v1.PersistentVolumeClaim {
-	pvcLabels := makeStorageClassDeviceSetPVCLabel(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, setIndex)
-
-	// pvc naming format <storageClassDeviceSetName>-<SetNumber>-<PVCIndex>
-	pvcGenerateName := pvcStorageClassDeviceSetPVCId + "-"
+func makeDeviceSetPVC(deviceSetName, pvcID string, setIndex int, pvcTemplate v1.PersistentVolumeClaim) *v1.PersistentVolumeClaim {
+	pvcLabels := makeStorageClassDeviceSetPVCLabel(deviceSetName, pvcID, setIndex)
 
 	// Add user provided labels to pvcTemplates
 	for k, v := range pvcTemplate.GetLabels() {
 		pvcLabels[k] = v
 	}
 
+	// pvc naming format rook-ceph-osd-<deviceSetName>-<SetNumber>-<PVCIndex>
 	return &v1.PersistentVolumeClaim{
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: pvcGenerateName,
-			Labels:       pvcLabels,
-			Annotations:  pvcTemplate.Annotations,
+			Name:        fmt.Sprintf("%s-%s", AppName, pvcID),
```

Comment:
Ok that's a good catch! To be safe to avoid re-using the same folder I will revert to generate the name again. It was just a nice-to-have for predictable names.